src/main/java/seedu/address/model/event/UniqueEventList.java

```diff
@@ -95,6 +113,28 @@ public void setEvents(List<Event> events) {
         internalList.setAll(events);
     }
 
+    /**
+     * Replaces the event list to remove {@code person} from contact list.
+     */
+    public void updateContacts(Person person) {
+        for (int i = 0; i < internalList.size(); i++) {
+            Event curr = internalList.get(i);
+            Set<Person> contactsList = curr.getContacts();
+            Set<Person> updatedContactsList = new HashSet<>();
+            if (contactsList.contains(person)) {
+                for (Person p : contactsList) {
+                    if (!p.equals(person)) {
+                        updatedContactsList.add(p);
+                    }
+                }
+            } else {
+                updatedContactsList = contactsList;
+            }
+            Event updatedEvent = new Event(curr.getName(), curr.getDate(), curr.getAddress(), updatedContactsList);
+            internalList.set(i, updatedEvent);
+        }
+    }
```

Comment:
ok now to think about it perhaps this one and the other `UpdateTag()` in `UniqueTagList.java` can be abstracted further to maintain SLAP but will KIV for now

Reply:
yup, we'll look into it in 1.3